ddl/ddl_api.go

```diff
@@ -3765,6 +3761,15 @@ func (d *ddl) getModifiableColumnJob(ctx sessionctx.Context, ident ast.Ident, or
 		return nil, errors.Trace(err)
 	}
 
+	// Check the column with foreign key, waiting for the default flen and decimal.
+	if fkInfo := getColumnForeignKeyInfo(originalColName.L, t.Meta().ForeignKeys); fkInfo != nil {
```

Comment:
Why we change the order of this code? Do we have a test to cover this update？

Reply:
The newCol.Flen was Unspecified until it is correctly assigned in `setCharsetCollationFlenDecimal` lined in 3760